src/amo/components/AddonMoreInfo.js

```diff
@@ -0,0 +1,75 @@
+import React, { PropTypes } from 'react';
+import { compose } from 'redux';
+
+import Link from 'amo/components/Link';
+import translate from 'core/i18n/translate';
+
+import './AddonMoreInfo.scss';
+
+
+export class AddonMoreInfoBase extends React.Component {
+  static propTypes = {
+    addon: PropTypes.object.isRequired,
+    i18n: PropTypes.object.isRequired,
+  }
+
+  render() {
+    const { addon, i18n } = this.props;
+
+    if (!addon) {
+      return null;
```

Comment:
Yeah, turns out I was being overly cautious; the parent component won't render without an `addon` anyway and it's required here too, so this is overkill. I moved the check up but it didn't even matter then, so I've just removed this.